### PR TITLE
net-firewall/ufw: bump to 0.36.1

### PR DESCRIPTION
**Pull request by hsntgm**
Package-Manager: Portage-3.0.30, Repoman-3.0.3
Signed-off-by: Hasan ÇALIŞIR <hasan.calisir@psauxit.com>

ebuild changelog 0.36.1:
* Version bump to 0.36.1
* EAPI 8
* python_combat drop 3.7, add 3.11, current range 8,11
* dropped NonexistentBlockers from RDEPEND
        !<kde-misc/kcm-ufw-0.4.2
	!<net-firewall/ufw-frontends-0.3.2"
* Removed RESTRICT="test" --> upstream bug: https://bugs.launchpad.net/ufw/+bug/815982 fixed and commited.
* Change mod 0644 for all rules in /etc/ufw
* Patches synced with current version that we still need them.

Regards.
~Hasan

### CODE REVIEW

**Comment by thesamesam** [comment]
Could you put the PR comment into the commit message too, so it doesn't get lost?

**Comment by hsntgm** [comment]
> Could you put the PR comment into the commit message too, so it doesn't get lost?
> 
> Thanks for doing this!

Done.

**Comment by Flowdalic** [comment]
Fails for me with

```
copying /var/tmp/portage/net-firewall/ufw-0.36.1/image/_python3.9/etc/ufw/user6.rules -> /var/tmp/portage/net-firewall/ufw-0.36.1/image/_python3.9/usr/share/ufw/iptables
 * Using python3.9 in global scope
 * python3_9: running distutils-r1_run_phase python_install_all
chmod: cannot access '/var/tmp/portage/net-firewall/ufw-0.36.1/image/etc/ufw/*.rules': No such file or directory
 * ERROR: net-firewall/ufw-0.36.1::gentoo failed (install phase):
 *   fperms failed
```

Full build log at http://0x0.st/oS5_.log

**Comment by hsntgm** [comment]
@Flowdalic let me check.

**Comment by Flowdalic** [comment]
I see that that you [dropped the `fperms` call](https://github.com/gentoo/gentoo/commit/442b612e4adebdbc8dc507c7dc0ffa720b72f93b). If there is a good reason to keep it, then the right, or at least one fix, according to [Gentoo bug #714082 comment #1](https://bugs.gentoo.org/714082#c1) would be

```bash
pushd "{ED}" || die
fperms -R 0644 etc/ufw/*.rules
popd || die
```

Do you want to re-add it?

**Comment by hsntgm** [comment]
> I see that that you [dropped the `fperms` call](https://github.com/gentoo/gentoo/commit/442b612e4adebdbc8dc507c7dc0ffa720b72f93b). If there is a good reason to keep it, then the right, or at least one fix, according to [Gentoo bug #714082 comment #1](https://bugs.gentoo.org/714082#c1) would be
> 
> ```shell
> pushd "{ED}" || die
> fperms -R 0644 etc/ufw/*.rules
> popd || die
> ```
> 
> Do you want to re-add it?

I WAS NOT SURE

File: net-firewall/ufw/ufw-0.36.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+	distutils-r1_python_prepare_all
+}
+
+python_install_all() {
+	newconfd "${FILESDIR}"/ufw.confd ufw
+	newinitd "${FILESDIR}"/ufw-2.initd ufw
+	systemd_dounit "${FILESDIR}/ufw.service"
+
+	pushd "{ED}" || die
```
Comment (Flowdalic) [inline net-firewall/ufw/ufw-0.36.1.ebuild:136]:
```suggestion
	pushd "${ED}" || die
```

**Review by Flowdalic — CHANGES_REQUESTED** [review]

**Comment by hsntgm** [comment]
@Flowdalic I understand that you are trying to help but please let me know your **on fly** changes next time. **Sorry but I hate this behaviour in gentoo community**. You pointed old package's patch to new one and dropped it I can accidently delete this patch when remove old ! I know QA says there is a duplicated patch but this is how I manage my package. 

@juippis
@mgorny 

* 	net-firewall/ufw: re-add dropped loong KEYWORD	Florian Schmaus
* 	net-firewall/ufw: use chmod instead of fperms	Florian Schmaus	
* 	net-firewall/ufw: remove duplicate bash completion patch	Florian Schmaus

```
--- ufw-0.36.1.ebuild   2022-07-01 10:56:03.251315401 +0300
+++ ufw-0.36.1.ebuild   2022-07-01 13:13:50.498175072 +0300
@@ -15,7 +15,7 @@

 LICENSE="GPL-3"
 SLOT="0"
-KEYWORDS="~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
+KEYWORDS="~amd64 ~arm ~arm64 ~ia64 ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86"
 IUSE="examples ipv6"

 RDEPEND="net-firewall/iptables[ipv6(+)?]"
@@ -29,7 +29,7 @@
        # Remove shebang modification.
        "${FILESDIR}/${P}-shebang.patch"
        # Fix bash completions, bug #526300
-       "${FILESDIR}/${P}-bash-completion.patch"
+       "${FILESDIR}/${PN}-0.36-bash-completion.patch"
 )

 pkg_pretend() {
@@ -134,7 +134,7 @@
        systemd_dounit "${FILESDIR}/ufw.service"

        pushd "${ED}" || die
-       fperms -R 0644 etc/ufw/*.rules
+       chmod -R 0644 etc/ufw/*.rules || die
        popd || die

        exeinto /usr/share/${PN}
```

**Comment by hsntgm** [comment]
@Flowdalic I didn't drop any keyword ~loong.
net-firewall/ufw: re-add dropped loong KEYWORD Florian Schmaus

I just realized that an other guy named **WANG Xuerui** added ~loong to my package without pinging me before my version bump. Then you re-added the KEYWORD like as it dropped by me !!

```
* 	net-firewall/ufw: keyword 0.36-r1 for ~loong	WANG Xuerui	3 days	1	-1/+1
``` 

**WHAT ARE YOU DOING GUYS??**
-----

**Project:Proxy Maintainers**

<html>
<body>


Lead(s) | Michał Górny (mgorny)
-- | --



</body>
</html>

@mgorny @thesamesam @juippis @monsieurp @tamiko @Zlogene

**Comment by floppym** [comment]
Arch teams do not typically inform individual maintainers when they add values to KEYWORDS.

I would suggest reviewing the gentoo repo for previous changes before you make any modifications. Attempting to stop developers from making trivial/obvious fixes is counter-productive.

**Comment by mgorny** [comment]
I don't understand your problem. Adding keywords is not that big a deal, and people don't coordinate much about it. Nobody blames you for loosing it, and the commit you referring to doesn't blame *you* either. It just states a fact that the keyword was dropped (i.e. lost), so it is being restored. There is no blame there, just a quick fix to save you the work and users the problem.

I don't understand why you are being so aggressive about it, and you claim that the project is "mismanaged". In fact, I feel insulted by these allegations.

**Comment by thesamesam** [comment]
It's normal to have to rebase before making PRs. Making small fixups post-merging generally is helpful as it avoids round-trips for small changes and delaying merging further.

Please be polite in future. I see your "symbolic commit" - I've dropped you as a proxied maintainer for now as this kind of language isn't acceptable.

**Pull request closed without merging**